src/renderers/testing/ReactShallowRenderer.js

```diff
@@ -0,0 +1,216 @@
+        );
+
+        if (element.type.hasOwnProperty('contextTypes')) {
+          ReactDebugCurrentFrame.element = element;
+
+          checkPropTypes(
+            element.type.contextTypes,
+            context,
+            'context',
+            getName(element.type, this._instance),
+            ReactDebugCurrentFrame.getStackAddendum,
+          );
+
+          ReactDebugCurrentFrame.element = null;
+        }
+
+        this._rendered = mountClassComponent(
+          this._instance,
+          element.props,
+          context,
+        );
+      } else {
+        this._rendered = element.type(element.props, context);
+      }
+    }
+
+    return this.getRenderOutput();
+  }
+
+  unmount() {
+    if (this._instance) {
+      if (typeof this._instance.componentWillUnmount === 'function') {
+        this._instance.componentWillUnmount();
+      }
+    }
+
+    this._context = null;
+    this._element = null;
+    this._newState = null;
+    this._rendered = null;
+    this._instance = null;
+  }
+}
+
+class Updater {
+  constructor(renderer) {
+    this._renderer = renderer;
+  }
+
+  isMounted(publicInstance) {
+    return !!this._renderer._element;
+  }
+
+  enqueueForceUpdate(publicInstance, callback, callerName) {
+    this._renderer.render(this._renderer._element, this._renderer._context);
+  }
+
+  enqueueReplaceState(publicInstance, completeState, callback, callerName) {
+    this._renderer._newState = completeState;
+    this._renderer.render(this._renderer._element, this._renderer._context);
+  }
+
+  enqueueSetState(publicInstance, partialState, callback, callerName) {
+    this._renderer._newState = {
+      ...publicInstance.state,
+      ...partialState,
+    };
+
+    this._renderer.render(this._renderer._element, this._renderer._context);
+  }
+}
+
+function getName(type, instance) {
+  var constructor = instance && instance.constructor;
+  return type.displayName ||
+    (constructor && constructor.displayName) ||
+    type.name ||
+    (constructor && constructor.name) ||
+    null;
+}
+
+function mountClassComponent(instance, props, context) {
+  instance.context = context;
+  instance.props = props;
+  instance.state = instance.state || emptyObject;
+
+  if (typeof instance.componentWillMount === 'function') {
+    instance.componentWillMount();
+  }
+
+  const rendered = instance.render();
+
+  if (typeof instance.componentDidMount === 'function') {
+    instance.componentDidMount();
+  }
+
+  return rendered;
+}
+
+function updateClassComponent(instance, rendered, props, state, context) {
+  state = state || emptyObject;
+
+  const oldContext = instance.context;
+  const oldProps = instance.props;
+  const oldState = instance.state;
+
+  if (typeof instance.componentWillReceiveProps === 'function') {
+    instance.componentWillReceiveProps(props);
+  }
+
+  if (typeof instance.shouldComponentUpdate === 'function') {
+    if (instance.shouldComponentUpdate(props, state, context) === false) {
+      return rendered;
```

Comment:
In addition to that, should shallow renderer update `state` and `context` as well?